### ISSUE TITLE
NRE on .As<T>

### ISSUE THREAD

**Issue by azchohfi** [microsoft/CsWinRT#490]
This code is throwing A NRE:
```cs
[ComImport]
[InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
[Guid("EECDBF0E-BAE9-4CB6-A68E-9598E1CB57BB")]
internal interface IWindowNative
{
    IntPtr WindowHandle { get; }
}

...

var window = new MainWindow();
IWindowNative windowWrapper = window.As<IWindowNative>();
var windowHandle = windowWrapper.WindowHandle;
```

Error:
```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=WinRT.Runtime
  StackTrace:
   at WinRT.CastExtensions.TryGetComposedRefForQI(Object value, IObjectReference& objRef)
   at WinRT.CastExtensions.TryGetRefForObject(Object value, Boolean allowComposed, IObjectReference& reference)
   at WinRT.CastExtensions.GetRefForObject(Object value)
   at WinRT.CastExtensions.As[TInterface](Object value)
```

**Comment by azchohfi** [comment]
@manodasanW Could you please tag this properly? Thanks!

**Comment by jkoritzinsky** [comment]
It looks like when we added `IDynamicInterfaceCastable` support, we accidentally removed the `GetDefaultReference` method that `TryGetComposedRefForQI` uses. It looks like we'll have to either add back the method or determine another way how to get the correct ref here. (Maybe extend `IWinRTObject` again?)

**Comment by Scottj1s** [comment]
fixed by #491
